### DISCUSSION TITLE
What is it I need to do to get a users profile info after successful Google OAuth?

### DISCUSSION THREAD

**Discussion by react-ghost** [supabase/supabase#5666] [Questions] [answered]
Hey there,

I just implemented google logins on my app utilizing Next.js and Google OAuth. 

I'm utilizing the code provided from the documentation.

```
const { user, session, error } = await supabase.auth.signIn({
  provider: 'google',
})
```

How am I supposed to extract user information FROM google profiles? I am currently triggering a function that inserts a new row with default values into my public.users table that at least allows the app to continue, but I'm confused as to how I would extract profile information from google itself. I'm seeing there's api endpoints I can call to get this it looks like, but even in that case I'm unsure as to how I'm supposed to know that it was a google user that signed in and to carry out that process, if that is in fact the best way to go about it.

**Comment by visudo20179C** [comment, marked as answer]
`supabase.auth.user()` should return the user object after sign in.

**Reply by react-ghost** [reply]
Just played around with it.. You're right, I guess I had just gotten used to handling all users information and putting it into my public.users table and I didn't realize the user object is sent with metadata that contains the information I'm looking for from google oauth.

Thank you.

**Reply by visudo20179C** [reply]
No problem. You can also use `supabase.auth.session()` to get the session data, if there is an active session.

**Reply by developerMahian** [reply]
In my supabase-react application, I'm using a typical database trigger to create a "profiles" row from the new user object.
Isn't there any way to access the meta_data from Google auth just before creating the user, so that I can manipulate and send the meta_data as payload while signing up?
